src/v0/destinations/pinterest_tag/transform.js

```diff
@@ -68,7 +68,7 @@ const commonFieldResponseBuilder = (
   const isValidUserPayload = checkUserPayloadValidity(userPayload);
   if (isValidUserPayload === false) {
     throw new InstrumentationError(
```

Comment:
We are using TransformationError in workflows and using InstrumentationError can cause comparison failures. @saikumarrs

Reply:
Yeah, `TransformationError` should be sparingly used. I'll fix this in a separate PR. Let's also deal with the comparison failures separately.